crates/uv-settings/src/lib.rs

```diff
@@ -206,3 +259,109 @@ pub enum Error {
     #[error("Failed to parse: `{0}`")]
     UvToml(String, #[source] toml::de::Error),
 }
+
+#[cfg(test)]
+mod test {
+    #[cfg(not(windows))]
+    use crate::locate_system_config_xdg;
+    #[cfg(windows)]
+    use crate::system_config_file;
+    #[cfg(windows)]
+    use uv_static::EnvVars;
+
+    use assert_fs::fixture::FixtureError;
+    use assert_fs::prelude::*;
+    use indoc::indoc;
+    use std::env;
+
+    #[test]
+    #[cfg(not(windows))]
+    fn test_locate_system_config_xdg() -> Result<(), FixtureError> {
+        // Write a `uv.toml` to a temporary directory.
+        let context = assert_fs::TempDir::new()?;
+        context.child("uv").child("uv.toml").write_str(indoc! {
+            r#"
+            [pip]
+            index-url = "https://test.pypi.org/simple"
+        "#,
+        })?;
+
+        let current_dir = env::current_dir().unwrap();
+
+        {
+            env::set_current_dir(context.path()).unwrap();
+
+            // None
+            assert_eq!(locate_system_config_xdg(None), None);
+
+            // Empty string
+            assert_eq!(locate_system_config_xdg(Some("")), None);
+
+            // Single colon
+            assert_eq!(locate_system_config_xdg(Some(":")), None);
+        }
+
+        env::set_current_dir(&current_dir).unwrap();
+        assert_eq!(locate_system_config_xdg(Some(":")), None);
+
+        // Assert that the `system_config_file` function returns the correct path.
+        assert_eq!(
+            locate_system_config_xdg(Some(context.to_str().unwrap())).unwrap(),
+            context.child("uv").child("uv.toml").path()
+        );
+
+        // Write a separate `uv.toml` to a different directory.
+        let first = context.child("first");
+        let first_config = first.child("uv").child("uv.toml");
+        first_config.write_str("")?;
+
+        assert_eq!(
+            locate_system_config_xdg(Some(
+                format!("{}:{}", first.to_string_lossy(), context.to_string_lossy()).as_str()
+            ))
+            .unwrap(),
+            first_config.path()
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    #[cfg(windows)]
+    fn test_windows_config() -> Result<(), FixtureError> {
+        // Write a `uv.toml` to a temporary directory.
+        let context = assert_fs::TempDir::new()?;
+        context
+            .child("ProgramData")
+            .child("uv")
+            .child("uv.toml")
+            .write_str(indoc! { r#"
+            [pip]
+            index-url = "https://test.pypi.org/simple"
+        "#})?;
+
+        // Previous value of `SYSTEMDRIVE` which should always exist.
+        let prev_system_drive = env::var(EnvVars::SYSTEMDRIVE).unwrap_or("C:".to_string());
+
+        // This is typically only a drive (that is, letter and colon) but we
+        // allow anything, including a path to the test fixtures...
+        env::set_var(EnvVars::SYSTEMDRIVE, context.path().as_os_str());
```

Comment:
Same as https://github.com/astral-sh/uv/pull/7851/files#r1808963874

Reply:
This is different than the above. The above is changing the current directory. This is changing an env var. (The same points may apply, I just wanted to point it out.)

Reply:
Yeah thanks for clarifying, the same problem is true unfortunately. If there's another test that reads `SYSTEMDRIVE` we can encounter a race condition and the tests will be flaky.

Reply:
Okay cool, got it!

Reply:
Unfortunately I think this may still be a problem even if you use `with_vars` because [temp-env](https://crates.io/crates/temp-env) creates a mutex to prevent concurrent conflicts between variables but if we're not using `temp-env` elsewhere we can still end up mutating their state. This isn't a big problem now because there aren't tests that rely on reading the environment that we're mutating in the Python discovery tests. The same is true of this test now, but it may not be true forever.